how-to/workspace-platform-starter/client/src/framework/workspace/notification-client.ts

```diff
@@ -40,6 +49,16 @@ export class NotificationClient implements NotificationClientInterface {
 		this._options = options;
 		this._idPrefix = options.idPrefix ?? `${options.id}-`;
 		this._platformId = platformId;
+		this._listenerRegister = [];
+        this._currentCount = 0;
+        this._logger = createLogger(`Notification Client (${this._idPrefix})`);
+        this.trackCount().then(() => {
```

Comment:
Because the track count can be expensive, instead of initiating it in the constructor, only start tracking it if someone subscribes to the notification-count-changed event in the addEventListener ?

Reply:
Done and added logic to clean up and stop if there are no listeners left.